### DISCUSSION TITLE
Create a pydantic.BaseModel definition from an external class or dictionary

### DISCUSSION THREAD

**Discussion by den4uk** [pydantic/pydantic#3055] [Question]
I have a (dynamic) definition of a simple class, like so:

```python
class Simple:
    val: int = 1
```

I intend to use this definition to build a `pydantic.BaseModel`, so it can be defined from the `Simple` class; basically doing this, but via `type`, under my own metaclass structure where the `Simple` class definition is retrieved from.

```python
from pydantic import BaseModel

class SimpleModel(Simple, BaseModel):
    pass

# Actual ways tried:
SimpleModel = type('SimpleModel', (Simple, BaseModel), {})
# or
SimpleModel = type('SimpleModel', (BaseModel, ), Simple.__annotations__)
```

However, this approach was not returning a model class with the parameters from the `Simple` class.

```python
>>> SimpleModel
{'title': 'SimpleModel', 'type': 'object', 'properties': {}}
```

I understand that the `BaseModel` already uses a rather complex metaclass under the hood, however, my intended implementation is also under a metaclass, where I intend to _dynamically_ transfer the `Simple` class into a `BaseModel` definition.

Your suggestions will be kindly appreciated.

**Comment by PrettyWood** [comment]
Hi @den4uk 
Here you go!
```py
from pydantic import create_model


class Simple:
    val: int = 1
    val_none: int = None
    val_no_default: str


field_definitions = {
    field_name: (field_type, getattr(Simple, field_name, ...))
    for field_name, field_type in Simple.__annotations__.items()
}
SimpleModel = create_model('SimpleModel', **field_definitions)
print(SimpleModel.__fields__)
# {
#   'val': ModelField(name='val', type=int, required=False, default=1),
#   'val_none': ModelField(name='val_none', type=Optional[int], required=False, default=None),
#   'val_no_default': ModelField(name='val_no_default', type=str, required=True),
# }
```
Hope it helps
